edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -40,29 +41,52 @@ class EnsureJWTAuthSettingsMiddleware(MiddlewareMixin):
     """
     _required_permission_classes = (NotJwtRestrictedApplication,)
 
+    @staticmethod
+    def _get_decomposed_permissions(permission):
+        """
+        Given a composed permission class or instance, return its component parts.
+
+        Examples:
+
+        Composition using rest condition:
+            C(IsStaff) | IsSuperuser -> [IsStaff, IsSuperuser]
+            C(IsOwner) | (C(IsStaff) | IsSuperuser) -> [IsOwner, (C(IsStaff) | IsSuperuser)]
+
+        Composition using DRF native support in 3.9+:
+            IsStaff | IsSuperuser -> [IsStaff, IsSuperuser]
+            IsOwner | IsStaff | IsSuperuser -> [IsOwner | IsStaff, IsSuperuser]
+
+        """
+        # We skip SingleOperandHolder since it contains negations
+        if isinstance(permission, OperandHolder):
+            return [permission.op1_class, permission.op2_class]
+        # Don't include negated permissions
+        if not getattr(permission, 'negated', False):
+            return getattr(permission, 'perms_or_conds', [])
```

Comment:
I suggest we raise a warning here to inform the use of a "deprecated" -ish library, hinting to be upgraded in a later release.

Reply:
I think on balance this code can just be removed, since using this release already means switching to a version of django that rest-condition doesn't support.